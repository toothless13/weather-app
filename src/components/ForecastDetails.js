const ForecastDetails = (props) => {
  const { date, temperature, humidity, wind } = props;
  const formattedDate = new Date(date).toDateString();
  const { min: minTemp, max: maxTemp } = temperature;
  const { speed: windSpeed, direction: windDirection } = wind;
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__min-temp">{minTemp}</div>
      <div className="forecast-details__max-temp">{maxTemp}</div>
      <div className="forecast-details__humidity">{humidity}</div>
      <div className="forecast-details__wind-speed">{windSpeed}</div>
      <div className="forecast-details__wind-direction">{windDirection}</div>
    </div>
  );
};

export default ForecastDetails;
