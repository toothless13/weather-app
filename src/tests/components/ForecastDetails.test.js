// eslint-disable-next-line no-unused-vars
import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails component", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 80,
    wind: {
      speed: 60,
      direction: "ne",
    },
  };

  it("Renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails
        date={validProps.date}
        temperature={validProps.temperature}
        humidity={validProps.humidity}
        wind={validProps.wind}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastDetails
        date={validProps.date}
        temperature={validProps.temperature}
        humidity={validProps.humidity}
        wind={validProps.wind}
      />,
    );

    expect(getByText("Thu Jan 01 1970")).toHaveAttribute(
      "class",
      "forecast-details__date",
    );
    expect(getByText("12")).toHaveAttribute(
      "class",
      "forecast-details__min-temp",
    );
    expect(getByText("22")).toHaveAttribute(
      "class",
      "forecast-details__max-temp",
    );
    expect(getByText("80")).toHaveAttribute(
      "class",
      "forecast-details__humidity",
    );
    expect(getByText("60")).toHaveAttribute(
      "class",
      "forecast-details__wind-speed",
    );
    expect(getByTestId("forecast-details__wind-direction")).toHaveAttribute(
      "class",
      "forecast-details__wind-direction",
    );
  });
});
