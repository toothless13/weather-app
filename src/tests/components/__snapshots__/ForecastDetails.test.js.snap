// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails component Renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__min-temp"
    >
      12
    </div>
    <div
      class="forecast-details__max-temp"
    >
      22
    </div>
    <div
      class="forecast-details__humidity"
    >
      80
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      60
    </div>
    <div
      class="forecast-details__wind-direction"
      data-testid="forecast-details__wind-direction"
    >
      ne
    </div>
  </div>
</DocumentFragment>
`;
